torchtune/_cli/download.py

```diff
@@ -159,3 +200,77 @@ def _download_cmd(self, args: argparse.Namespace) -> None:
             *list(Path(true_output_dir).iterdir()),
             sep="\n",
         )
+
+    def _download_from_kaggle(self, args: argparse.Namespace) -> None:
+        """Downloads a model from the Kaggle Model Hub."""
+
+        # Note: Kaggle doesn't actually use the "repository" terminology, but we still reuse args.repo_id here for simplicity
+        model_handle = args.repo_id
+        self._validate_kaggle_model_handle(model_handle)
+
+        # kagglehub doesn't currently support `local_dir` like huggingface_hub
+        if args.output_dir is not None:
+            warn("--output-dir flag is not supported for Kaggle model downloads")
+
+        if args.kaggle_username is not None and args.kaggle_api_key is not None:
+            set_kaggle_credentials(args.kaggle_username, args.kaggle_api_key)
+        elif self._xor(
```

Comment:
nit: I think its clearer if I saw something like below.

```python
if args.kaggle_username is None and args.kaggle_api_key is None:
    warn("username and api key are not set")
elif args.kaggle_username and args.kaggle_api_key is None:
    warn("api key is not set")
elif args.kaggle_username is None and args.kaggle_api_key:
    warn("username is not set")

 set_kaggle_credentials(args.kaggle_username, args.kaggle_api_key)
```

Reply:
I split this up over 3 `if` checks. But rather than `warn` I raise the error. I went this route since `set_kaggle_credentials` throws anyway when one of the provided arguments is empty. Throwing the error before `set_kaggle_credentials` allows for a clearer error message that references the required CLI args.

also, for the `args.kaggle_username is None and args.kaggle_api_key is None` case, I didn't want to throw here since we should still support anonymous downloads for public, non-gated models.